### DISCUSSION TITLE
Speeding up silencedetect

### DISCUSSION THREAD

**Discussion by Octavio-T** [lagmoellertim/unsilence#103] [Ideas]
Hello, I'm quite inexperienced when it comes to coding and Git, but I've done work with Ffmpeg for archiving online class recordings for quite a while since online classes became a thing.

You can speed up the silence detect part by not including the video using the `-vn` argument of Ffmpeg. I've added it in this file: [DetectSilence.txt](https://github.com/lagmoellertim/unsilence/files/6900183/DetectSilence.txt)

This should cut down on detection time significantly, since silencedetect only needs the audio.

I don't know how to do stuff in Git, so I don't think I can do much when it comes to hardware support like Intel qsv and Nvidia nvec.

I hope this helped.

**Comment by lagmoellertim** [comment]
Hey, I tested the `-vn` flag, but it doesn't seem to improve speed (tested it with a large video file, the process took 1 minute with and without the flag).

Have you tested this before and got other results?

**Reply by Octavio-T** [reply]
Apologies, the file I sent was not modified. Must have overlooked while testing.


The `-vn` flag seems to have sped up silencedetect on my machine. I'm running on a very low-powered Windows device.

This is what the line in DetectSilence.py looked like after modifying:
```
command = [
        "ffmpeg",
        "-i", str(input_file),
        "-vn",
        "-af",
        f"silencedetect=noise={kwargs.get('silence_level', -35)}dB:d={kwargs.get('silence_time_threshold', 0.5)}",
        "-f", "null",
        "-"
    ]
```

Since ffmpeg is used, I measured the execution time for ffmpeg to process the whole video directly on Powershell. The video I'm testing is a 3-hour long class recording.

These are the results for my machine.

`Measure-Command {ffmpeg -hide_banner -i .\01.mp4 -af silencedetect -f null -}`
![image](https://user-images.githubusercontent.com/88143242/127543571-546a1444-13df-482f-ab23-b929dc8bf1b4.png)

`Measure-Command {ffmpeg -hide_banner -i .\01.mp4 -vn -af silencedetect -f null -}`
![image](https://user-images.githubusercontent.com/88143242/127543366-18cc8b62-d1d5-47a1-8dc1-8c7ce25cb8f1.png)

**Reply by Jorge339** [reply]
Hi, I can confirm, I tried this fix myself and it greatly increased detection time, just replacing the code above into the DetectSilence.py and that's it, here is the ready to use file:
[DetectSilence.txt](https://github.com/lagmoellertim/unsilence/files/7151926/DetectSilence.txt)
